### PR TITLE
NAS-128911 / 24.10 / Add basic tests for statx wrapper

### PR DESCRIPTION
**Pull request by anodos325**
* Validate that output matches what we see from os.stat

* Validate that opening with dirfd works as expected

* Validate that opening with AT_EMPTY_PATH works as expected

### CODE REVIEW

File: src/middlewared/middlewared/pytest/unit/utils/test_statx.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    'DEV',
+    'INO',
+    'SIZE',
+    'BLOCKS',
+    'BLKSIZE',
+    'NLINK',
+]
+
+def timespec_convert(timespec):
+    return timespec.tv_sec + timespec.tv_nsec / 1000000000
```
Comment (bmeagherix) [inline src/middlewared/middlewared/pytest/unit/utils/test_statx.py:24]: Hmmm.  What are you trying to do with this calculation?  **round** to nearest second, **floor** to the second?

Either way, presumably `timespec.tv_nsec / 1000000000` will always be < 1

AFAICS `os.stat` yields a whole int.

File: src/middlewared/middlewared/pytest/unit/utils/test_statx.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        case 'NLINK':
+            assert st1.st_nlink == st2.stx_nlink
+        case 'SIZE':
+            assert st1.st_size == st2.stx_size
+        case _:
+            raise ValueError(f'{stat_prop}: unknown stat property')
+
+
+@pytest.mark.parametrize('stat_prop', BASIC_STAT_ATTRS)
+def test__check_statx_vs_stat_file(tmpdir, stat_prop):
```
Comment (bmeagherix) [inline src/middlewared/middlewared/pytest/unit/utils/test_statx.py:66]: Err, what's `tmpdir`?

File: src/middlewared/middlewared/pytest/unit/utils/test_statx.py (hunk 1 above)
Comment (bmeagherix) [reply src/middlewared/middlewared/pytest/unit/utils/test_statx.py:24]:
```
>>> os.stat('/tmp')
os.stat_result(st_mode=17407, st_ino=1, st_dev=44, st_nlink=9, st_uid=0, st_gid=0, st_size=180, st_atime=1715354418, st_mtime=1715357642, st_ctime=1715357642)
>>> os.stat('/tmp').st_mtime
1715357642.7296498
>>>
```
Weird, but OK

File: src/middlewared/middlewared/pytest/unit/utils/test_statx.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    st1, st2 = do_stat(os.path.join(tmpdir, 'testfile'), False)
+    validate_stat(stat_prop, st1, st2)
+
+
+@pytest.mark.parametrize('stat_prop', BASIC_STAT_ATTRS)
+def test__check_statx_vs_stat(tmpdir, stat_prop):
```
Comment (mgrimesix) [inline src/middlewared/middlewared/pytest/unit/utils/test_statx.py:72]: Consistent naming convention might be nice.
e.g. `test__check_file_statx_vs_stat` and `test__check_file_statx_vs_stat`

**Review by bmeagherix — APPROVED** [review]

**Review by mgrimesix — APPROVED** [review]

**Pull request merged**